subworkflows/local/profiling.nf

```diff
@@ -384,7 +384,22 @@ workflow PROFILING {
     }
 
     if ( params.run_krakenuniq ) {
-        ch_input_for_krakenuniq =  ch_input_for_profiling.krakenuniq
+        // Collect channel into list.  Sort to ensure batch membership remains constant across runs.
+        // This will enable retrieval of cached tasks.  This is a blocking operation.
+        ch_input_for_krakenuniq_sorted = ch_input_for_profiling.krakenuniq
+            .collect(
+                flat: false,
+                sort: {
+                    a,b ->  a[0].id <=> b[0].id ?:
+                            a[0].run_accession <=> b[0].run_accession ?:
+                            a[0].db_meta.db_name <=> b[0].db_meta.db_name ?:
+                            a[0].db <=> b[0].db
+                }   
+            )   
+            // Apply inverse of collect operator.  Result is multi-value channel.
+            .flatMap { it -> it.toList() }
```

Comment:
If I predict correctly what happens in the code, the output of the `collect` operator is a channel with a single element that is a list. If that holds, isn't it enough to just apply `flatMap` without the closure as you have done here?